rust/abacus-base/src/settings/chains.rs

```diff
@@ -54,53 +49,74 @@ pub struct GelatoConf {
 /// Addresses for outbox chain contracts
 #[derive(Clone, Debug, Deserialize, Default)]
 #[serde(rename_all = "camelCase")]
-pub struct OutboxAddresses {
-    /// Address of the Outbox contract
-    pub outbox: String,
+pub struct CoreContractAddresses {
+    /// Address of the mailbox contract
+    pub mailbox: String,
+    /// Address of the MultisigIsm contract
+    pub multisig_ism: String,
     /// Address of the InterchainGasPaymaster contract
-    pub interchain_gas_paymaster: Option<String>,
+    pub interchain_gas_paymaster: String,
 }
 
-/// Addresses for inbox chain contracts
-#[derive(Clone, Debug, Deserialize, Default)]
+/// Outbox indexing settings
+#[derive(Debug, Deserialize, Default, Clone)]
 #[serde(rename_all = "camelCase")]
-pub struct InboxAddresses {
-    /// Address of the Inbox contract
-    pub inbox: String,
-    /// Address of the InboxValidatorManager contract
-    pub validator_manager: String,
+pub struct IndexSettings {
+    /// The height at which to start indexing the Outbox contract
+    pub from: Option<String>,
+    /// The number of blocks to query at once at which to start indexing the
+    /// Outbox contract
+    pub chunk: Option<String>,
+}
+
+impl IndexSettings {
+    /// Get the `from` setting
+    pub fn from(&self) -> u32 {
+        self.from
+            .as_ref()
+            .and_then(|s| s.parse::<u32>().ok())
+            .unwrap_or_default()
+    }
+
+    /// Get the `chunk_size` setting
+    pub fn chunk_size(&self) -> u32 {
+        self.chunk
+            .as_ref()
+            .and_then(|s| s.parse::<u32>().ok())
+            .unwrap_or(1999)
+    }
 }
 
 /// A chain setup is a domain ID, an address on that chain (where the outbox or
 /// inbox is deployed) and details for connecting to the chain API.
 #[derive(Clone, Debug, Deserialize, Default)]
 #[serde(rename_all = "camelCase")]
-pub struct ChainSetup<T> {
+pub struct ChainSetup {
     /// Chain name
     pub name: String,
     /// Chain domain identifier
     pub domain: String,
     /// Number of blocks until finality
     pub finality_blocks: String,
     /// Addresses of contracts on the chain
-    pub addresses: T,
+    pub addresses: CoreContractAddresses,
     /// The chain connection details
     #[serde(flatten)]
     pub chain: ChainConf,
     /// How transactions to this chain are submitted.
     #[serde(default)]
     pub txsubmission: TransactionSubmissionType,
-    /// Set this key to disable the inbox. Does nothing for outboxes.
-    #[serde(default)]
-    pub disabled: Option<String>,
```

Comment:
I was thinking that you would only pass in config for the chains that you actually wanted to use.

So chain enabling/disabling would happen at the config JSON file level, and then again at the agent level with `chain_names` (so that validators can ignore every chain but one).

Happy to add it back in if you feel another layer of control is useful

Reply:
Ah I see, that definitely works but is a bit more awkward given the way we ship configs, where we have the environment-specific JSON file that defines basic info on all the chains in that environment, and then one can use environment variables to make additions to that JSON. With the `disabled` property it's really easy to disable a chain by just supplying the appropriate env var, but if you instead need to make changes to the JSON yourself then it's more involved

I'm leaning more toward adding `disabled` back, even though I agree that changing the JSON feels like a more correct approach.
Another option could be to pass in the enabled chains to the relayer settings

Reply:
Shouldn't this config for the relayer live in the whitelist/blacklist?

I'm going to keep removed for now and we can add back in later if necessary